### PR TITLE
add optional gallery submissions on prompts

### PR DESCRIPTION
**Pull request by ScuffedNewt**

### CODE REVIEW

File: app/Http/Controllers/Users/SubmissionController.php [hunk 1]
```diff
@@ -86,21 +87,31 @@ public function getNewSubmission(Request $request) {
         $closed = !Settings::get('is_prompts_open');
         $inventory = UserItem::with('item')->whereNull('deleted_at')->where('count', '>', '0')->where('user_id', Auth::user()->id)->get();
 
+        if (config('lorekeeper.settings.allow_gallery_submissions_on_prompts')) {
+            $gallerySubmissions = GallerySubmission::where('user_id', Auth::user()->id)->where('status', 'Accepted')->orderBy('id', 'DESC')->get()->pluck('title', 'id');
```
Comment (itinerare) [inline app/Http/Controllers/Users/SubmissionController.php:91]: Would be worth including submissions on which the user collaborated as well.

File: app/Models/Submission/Submission.php [hunk 2]
```diff
@@ -38,7 +39,7 @@ class Submission extends Model {
      * @var array
      */
     public static $createRules = [
-        'url' => 'nullable|url',
+        'url' => 'nullable',
```
Comment (itinerare) [inline app/Models/Submission/Submission.php:42]: The `url` validation rule was added as part of a way to harden that field... is there a reason it's removed?

**Review by itinerare — CHANGES_REQUESTED** [review]
Broadly looks good!

**Review by SpeedyD — APPROVED** [review]
Approved on the condition @itinerare's comments are taken into consideration. :)

File: app/Models/Submission/Submission.php (hunk 2 above)
Comment (Ne-wt) [reply app/Models/Submission/Submission.php:42]: in order for it to be used as a `title` field if needs be, but I can always just add it back!

File: app/Http/Controllers/Users/SubmissionController.php (hunk 1 above)
Comment (Ne-wt) [reply app/Http/Controllers/Users/SubmissionController.php:91]: true!

File: app/Models/Submission/Submission.php (hunk 2 above)
Comment (itinerare) [reply app/Models/Submission/Submission.php:42]: Hmm. I think it'd probably be safest to add it back, though open to other solutions that achieve the desired functionality *and* safety.

**Review by itinerare — CHANGES_REQUESTED** [review]

**Review by itinerare — APPROVED** [review]

**Pull request merged**